ambry-api/src/main/java/com/github/ambry/store/MessageInfo.java

```diff
@@ -295,4 +295,175 @@ public String toString() {
         .append("]");
     return stringBuilder.toString();
   }
+
+  /**
+   * A builder class for {@link MessageInfo}.
+   */
+  public static class Builder {
```

Comment:
I personally prefer to have Builder as a public static class of what it's building. I think this makes more sense since if the MessageInfo doesn't exist, this builder would not exist either.